docs/content/sips/017-service-chaining.md

```diff
@@ -14,30 +14,26 @@ Created: Feb 21, 2024
 
 # User experience
 
 ## Expressing a requirement for chaining
 
-To express that the application requires chaining for self-requests, add the following to `spin.toml`:
+To express that the application requires service chaining, add the following to `spin.toml`:
 
 ```toml
-[application.prerequisites]
-service_chaining = "required"
+[application.host_requirements]
+spin_internal_requests = "required"
 ```
 
-Entries in the `prerequisites` table can have any name, but the value must be one of:
-
-* `"required"` - the host must reject the application if it does not recognise the name, or cannot fulfil the requirement
-* `"preferred"` - the host should warn if it does not recognise the name, or cannot fulfil the requirement, but it shouldn't reject the application
-* `"optional"` (default) - this is the same as omitting the requirement but it might be nice to have a way to state it explicitly
+> Why ask the user to write out `"required"` instead of just listing the names?  In future, we may have features where it makes sense to say "I'd prefer to run on a host that supports this, but it's not a blocker," or where the user can specify a value (for a bad example, consider an "available memory" prerequisite).
 
-**BIKESHED:** Or something more explicit such as `direct_self_requests`? I'm not sure if "service chaining" will be obvious to readers (it wasn't to me the first time I heard it).
+If a host doesn't understand the `host_requirements` section, it must reject the application.  We can't retro-fit this to existing hosts so we have to rely on `serde` for this, which means the error will be unlovely.  However, for individual fields within the section, hosts can give better errors/warnings.
 
-**BIKESHED:** `prerequisites` is a pretty grim name. Other ideas: `requirements` or `features` (maybe too generic), or something referring to the host e.g. er `host` or `host_requirements`, which may be close the door to other kinds of precondition but is at least explicit.
-
-If a host doesn't understand the `prerequisites` section, it must reject the application.  We can't retro-fit this to existing hosts so we have to rely on `serde` for this, which means the error will be unlovely.  However, for individual fields within the section, hosts can give better errors/warnings.
+If a host understands the section, but doesn't understand an entry in that section, it _must_ reject the application with an error message explaining the required host features.
```

Comment:
Should we bake in a non-mandatory value now (`"preferred"`/`"optional"`)? These could still be disallowed for this specific feature but would permit backward-compatible additions in the future.